modules/apps/commerce/commerce-pricing-web/src/main/java/com/liferay/commerce/pricing/web/internal/frontend/taglib/servlet/taglib/CPInstancePriceEntriesScreenNavigationEntry.java

```diff
@@ -0,0 +1,137 @@
+import com.liferay.portal.kernel.security.permission.resource.ModelResourcePermission;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.io.IOException;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Pedro Victor Silvestre
+ */
+@Component(
+	property = "screen.navigation.entry.order:Integer=10",
+	service = ScreenNavigationEntry.class
+)
+public class CPInstancePriceEntriesScreenNavigationEntry
+	extends CPInstancePriceEntriesScreenNavigationCategory
+	implements ScreenNavigationEntry<CommercePriceEntry> {
+
+	@Override
+	public String getEntryKey() {
+		return getCategoryKey();
+	}
+
+	@Override
+	public boolean isVisible(User user, CommercePriceEntry commercePriceEntry) {
+		if (commercePriceEntry == null) {
+			return false;
+		}
+
+		boolean hasPermission = false;
+
+		try {
+			hasPermission = _commercePriceListModelResourcePermission.contains(
+				PermissionThreadLocal.getPermissionChecker(),
+				commercePriceEntry.getCommercePriceListId(), ActionKeys.VIEW);
+		}
+		catch (Exception exception) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(exception);
+			}
+		}
+
+		return hasPermission;
+	}
+
+	@Override
+	public void render(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse)
+		throws IOException {
+
+		try {
+			CPInstanceCommercePriceEntryDisplayContext
+				cpInstanceCommercePriceEntryDisplayContext =
+					new CPInstanceCommercePriceEntryDisplayContext(
+						_actionHelper, _commercePriceEntryService,
+						_commercePriceListActionHelper, httpServletRequest,
+						_itemSelector);
+
+			httpServletRequest.setAttribute(
+				WebKeys.PORTLET_DISPLAY_CONTEXT,
+				cpInstanceCommercePriceEntryDisplayContext);
+		}
+		catch (Exception exception) {
+			_log.error(exception);
+		}
+
+		_jspRenderer.renderJSP(
+			_setServletContext, httpServletRequest, httpServletResponse,
+			"/commerce_price_lists/cp_instance/details.jsp");
+	}
+
+	private static final Log _log = LogFactoryUtil.getLog(
+		CPInstancePriceEntriesScreenNavigationEntry.class);
+
+	@Reference
+	private ActionHelper _actionHelper;
+
+	@Reference
+	private CommercePriceEntryService _commercePriceEntryService;
+
+	@Reference
+	private CommercePriceListActionHelper _commercePriceListActionHelper;
+
+	@Reference(
+		target = "(model.class.name=com.liferay.commerce.price.list.model.CommercePriceList)"
+	)
+	private ModelResourcePermission<CommercePriceList>
+		_commercePriceListModelResourcePermission;
+
+	@Reference
+	private ItemSelector _itemSelector;
+
+	@Reference
+	private JSPRenderer _jspRenderer;
+
+	@Reference(
+		target = "(osgi.web.symbolicname=com.liferay.commerce.pricing.web)"
+	)
+	private ServletContext _setServletContext;
```

Comment:
@daiazius Please append an SF commit to rename all `_setServletContext` in this PR to `_servletContext`. Please also double check your other PRs, I think it may be common in Commerce.